macOS runners keep hanging

## Summary
<!-- One sentence summary of the issue -->

In the past week, the macOS test runners have been consistently hanging on the "Configure SSH" step:

https://github.com/OpenBagTwo/EnderChest/blob/0f7b4dae60dffd444e9939b9f4969d0d4a172fbd/.github/workflows/pull_request.yml#L37-L47

## Steps to Reproduce
<!--Describe in detail how to reproduce the issue. Include any commands you're running and any error messages or stack traces they generate. For long stack traces, please make use of the <details> tag: https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d -->

Open a PR

## Workaround
<!-- Is there a workaround for the issue? If so, post it here, as it may help with implementing a bugfix -->

If I cancel and rerun the failed jobs, they'll sometimes go through

## Severity
<!-- How painful is this bug? Example severity levels (feel free to choose your own and be as detailed as you like)

- Trivial
- Annoying
- Painful
- Gamebreaking
- World-ending (oh no!)
-->

Painful

## Desired Outcome
<!-- Examples:
- research
- fix
- go to your room and think about what you did
-->

1. **Research** if other people are reporting this problem and have insights into its cause.
1. If a **fix** is available, apply that fix and consider switching back to an upstream provider for the action

### Alternatives
- Don't configure local SSH at all on the macOS test runners and instead rely on the Linux workers to run the `--use-local-ssh` (non-cached) variant of the tests

## Urgency
<!-- When should this be fixed? Examples:
- None (for QoL issues or things that are unlikely to affect real-world use)
- By Next Major Release (either because the fix would require significant effort to resolve or because it would break backwards compatibility)
- By Next Minor Release (it's fine to bundle the fix in with other changes, but the next time this package gets an update, this should be resolved)
- Emergency (a patch needs to be pushed out ASAP)
-->

ASAP--the workaround is painful

## Notes
<!-- If you've done any additional research, please include a summary of your findings here -->
This issue originally mistakenly attributed the hanging to the subsequent "install rsync" step

Okay, so it's definitely hanging at the "verify that SSH is actually working" step:

```
2023-11-06T16:54:08.3802390Z +----[SHA256]-----+
2023-11-06T16:54:08.3918000Z SSH configured for local passwordless authentication
2023-11-06T17:01:08.2641470Z ##[error]The operation was canceled.
2023-11-06T17:01:08.2886070Z Post job cleanup.
2023-11-06T17:01:08.4880470Z [command]/usr/local/bin/git version
```

which makes me think I might be able to fix it with timeouts and retries.

The issue was DNS resolution:

```
2023-11-06T17:06:32.1855970Z ssh: Could not resolve hostname mac-1699290219374.local: nodename nor servname provided, or not known
2023-11-06T17:06:32.1878640Z ##[error]Process completed with exit code 255.
```

:facepalm: 

This should be fixed by de6af69bf6178c68f3c41fdcda025d8545080db0 which literally just changes "$HOSTNAME" to "localhost"

--- a/.github/workflows/pull_request.yml
+++ b/.github/workflows/pull_request.yml
@@ -44,6 +44,7 @@ jobs:
         EOF
         cat ~/.ssh/id_ed25519.pub > ~/.ssh/authorized_keys
         chmod -R 700 ~/.ssh
+        echo "SSH configured for local passwordless authentication"
         ssh -o 'StrictHostKeyChecking no' $USER@$HOSTNAME "echo It works"
     - name: Install Rsync
       if: matrix.os == 'macos-latest'

--- a/.github/workflows/pull_request.yml
+++ b/.github/workflows/pull_request.yml
@@ -44,8 +44,10 @@ jobs:
         EOF
         cat ~/.ssh/id_ed25519.pub > ~/.ssh/authorized_keys
         chmod -R 700 ~/.ssh
-        echo "SSH configured for local passwordless authentication"
-        ssh -o 'StrictHostKeyChecking no' $USER@$HOSTNAME "echo It works"
+    - name: Verify Local SSH
+      if: matrix.os != 'windows-latest'
+      run: |
+        ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 -o BatchMode=yes $USER@$HOSTNAME "echo It works"
     - name: Install Rsync
       if: matrix.os == 'macos-latest'
       run: brew install rsync

--- a/.github/workflows/pull_request.yml
+++ b/.github/workflows/pull_request.yml
@@ -47,7 +47,7 @@ jobs:
     - name: Verify Local SSH
       if: matrix.os != 'windows-latest'
       run: |
-        ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 -o BatchMode=yes $USER@$HOSTNAME "echo It works"
+        ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 -o BatchMode=yes $USER@localhost "echo It works"
     - name: Install Rsync
       if: matrix.os == 'macos-latest'
       run: brew install rsync

--- a/enderchest/cli.py
+++ b/enderchest/cli.py
@@ -550,6 +550,7 @@ def generate_parsers() -> tuple[ArgumentParser, dict[str, ArgumentParser]]:
         sync_parser.add_argument(
             "--exclude",
             "-e",
+            action="extend",
             nargs="+",
             help="Provide any file patterns you would like to skip syncing",
         )

--- a/enderchest/test/test_cli.py
+++ b/enderchest/test/test_cli.py
@@ -493,6 +493,42 @@ class TestOpen:
             "verbosity": 0,
         }
 
+    def test_passing_in_multiple_exclude_flags(self, monkeypatch) -> None:
+        sync_log: list[tuple[str, str, dict]] = []
+
+        def mock_sync(root, op, **kwargs) -> None:
+            sync_log.append((root, op, kwargs))
+
+        monkeypatch.setattr(remote, "sync_with_remotes", mock_sync)
+
+        action, root, _, kwargs = cli.parse_args(
+            [
+                "enderchest",
+                *self.action.split(),
+                "--exclude",
+                "*.secret",
+                "--dry-run",
+                "-e",
+                "private",
+                "super-private",
+                "-w",
+                "6",
+                "--exclude",
+                "do-not-sync",
+                "-vv",
+            ]
+        )
+        action(root, **kwargs)
+
+        assert len(sync_log) == 1
+        assert sync_log[0][2] == {
+            "dry_run": True,
+            "sync_confirm_wait": 6,
+            "exclude": ["*.secret", "private", "super-private", "do-not-sync"],
+            "verbosity": 2,
+            "timeout": None,
+        }
+
 
 class TestClose(TestOpen):
     action = "close"

